packages/atlas/src/views/viewer/VideoView/CommentThread.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useState } from 'react'
+
+import { useComments } from '@/api/hooks'
+import { CommentOrderByInput } from '@/api/queries'
+import { Comment, CommentProps } from '@/components/_comments/Comment'
+import { CommentInput } from '@/components/_comments/CommentInput'
+import { ReactionId } from '@/config/reactions'
+import { absoluteRoutes } from '@/config/routes'
+import { useReactionTransactions } from '@/hooks/useReactionTransactions'
+import { useUser } from '@/providers/user'
+
+import { getCommentReactions } from './utils'
+
+type CommentThreadProps = {
+  onOpenSignInDialog: () => void
+  commentId: string
+  videoId?: string
+  onEditLabelClick: () => void
+  videoAuthorId?: string
+  handleCancelConfirmation: (cb: () => void) => void
+  handleCommentReaction: (commentId: string, reactionId: ReactionId) => void
+} & CommentProps
+
+export const CommentThread: React.FC<CommentThreadProps> = ({
+  onOpenSignInDialog,
+  commentId,
+  videoId,
+  onEditLabelClick,
+  videoAuthorId,
+  handleCancelConfirmation,
+  handleCommentReaction,
+  ...commentProps
+}) => {
+  const [repliesOpen, setRepliesOpen] = useState(false)
+  const [replyInputOpen, setReplyInputOpen] = useState(false)
+  const [commentBody, setCommentBody] = useState('')
+  const [highlightedComment, setHighlightedComment] = useState<string | null>(null)
+  const { activeMemberId, activeMembership } = useUser()
+  const { processingCommentReactionId, addComment, commentInputProcessing } = useReactionTransactions()
+  const { comments, loading } = useComments(
+    { where: { parentComment: { id_eq: commentId } }, orderBy: CommentOrderByInput.CreatedAtAsc },
+    { skip: !commentId }
+  )
+  const placeholderItems = loading && !comments ? Array.from({ length: 4 }, () => ({ id: undefined })) : []
+  const replyAvatars = comments?.map((comment) => ({
+    url: comment?.author.metadata.avatar?.__typename === 'AvatarUri' ? comment?.author.metadata.avatar?.avatarUri : '',
+    handle: comment.author.handle,
+  }))
+
+  const toggleRepliesOpen = () => {
+    setRepliesOpen((prevState) => {
+      if (prevState) {
+        setReplyInputOpen(false)
+      }
+      return !prevState
+    })
+  }
+
+  const handleCancel = () => {
+    setReplyInputOpen(false)
+    setCommentBody('')
+  }
+
+  const handleComment = useCallback(
```

Comment:
wrapping this in useCallback is unnecessary because where it's used the handler is created every render anyways